### PR TITLE
feat: Add docker-compose to indexer

### PR DESCRIPTION
**Pull request by roninjin10**
- adds a docker-compose file for running indexer with docker-compose up
- adds example.env file

### CODE REVIEW

File: indexer/example.env [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Fill me in with goerli uris and run docker-compose up to run indexer vs goerli
+INDEXER_L1_ETH_RPC=FILL_ME_IN
+INDEXER_L2_ETH_RPC=FILL_ME_IN
+
```
Comment (hamdiallam) [inline indexer/example.env:4]: rm new line

**Review by hamdiallam — APPROVED** [review]
Generally LGTM! I'll play around with this on my machine

File: indexer/example.env (hunk 1 above)
Comment (roninjin10) [reply indexer/example.env:4]:
```suggestion
```

File: indexer/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+See the flags in `flags.go` for reference of what command line flags to pass to `go run`
+
+### Run indexer vs devnet
+
+TODO add indexer to the optimism devnet compose file (previously removed for breaking CI)
+
+### Run indexer vs a custom configuration
+
+`docker-compose.dev.yml` is git ignored.   Fill in your own docker-compose file here.
+
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply indexer/README.md:23]:
```suggestion
```

File: indexer/.gitignore [hunk 3]
```diff
@@ -0,0 +1,3 @@
+docker-compose.dev.yml
+.env
+
```
[the remark this answers is not in the document]
Comment (roninjin10) [reply indexer/.gitignore:3]:
```suggestion
```

**Pull request merged**